frontend/src/components/DeviceShareContainer/DeviceShareContainer.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react'
+import { IService } from 'remote.it'
+import { SharingDetails } from './SharingForm'
+import { useParams, useLocation, useHistory } from 'react-router-dom'
+import { SharingManager } from '../../services/SharingManager'
+import { makeStyles } from '@material-ui/core/styles'
+import { DeviceShareContent } from './DeviceShareContent'
+import { useDispatch, useSelector } from 'react-redux'
+import { Dispatch, ApplicationState } from '../../store'
+
+export const DeviceShareContainer = ({ username = '' }) => {
+  const { deviceID = '' } = useParams()
+  const { devices } = useDispatch<Dispatch>()
+  const { myDevice } = useSelector((state: ApplicationState) => ({
+    myDevice: state.devices.all.find(device => device.id === deviceID),
+  }))
+
+  const history = useHistory()
+  const location = useLocation()
+  const css = useStyles()
+
+  const handleShare = async (share: SharingDetails) => {
+
+    const shareData = mapShareData(share)
+    const contacts = share.contacts
+    const {scripting, services} = share.access
+
+    await SharingManager.share({...shareData, contacts})
```

Comment:
This is nice and cleanly done, but a little unconventional. Usually we do the async calls in the redux model's effects. Maybe we should discuss? We could keep the component focused on the ui this way.